### DISCUSSION TITLE
Compilation error caused because of unresolved spec target error in object.spec.move

### DISCUSSION THREAD

**Discussion by shivam2320** [aptos-labs/aptos-developer-discussions#467] [Questions] [answered]
### Describe your question in detail.

I'm using the `mainnet` version of `AptosFramework` but I'm unable to compile modules due to an unresolved spec target error in object.spec.move.

<img width="821" alt="Screenshot 2024-10-02 at 10 25 11 PM" src="https://github.com/user-attachments/assets/15f4d1b8-6d4a-4c77-8e9e-f52c7cb554ee">

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

**Comment by JohnChangUK** [comment]
Can you try overriding the standard library version by `mainnet/testnet/devnet`

`aptos move compile --override-std <OVERRIDE_STD>`
- [possible values: mainnet, testnet, devnet]

Note: Do you mind expanding on the fact that your dependencies are using different versions of `aptos-core`, hence why it is causing the compilation issue?

**Comment by shivam2320** [comment, marked as answer]
Hey I just updated the Aptos CLI and everything seems to be working fine now
